src/main/java/org/example/Main.java

```diff
@@ -121,339 +177,256 @@ public boolean isHappy() {
-                            continue;
-                        }
-                        if (users.contains("PALINDROMIC") && !num.palindromic) {
-                            continue;
-                        }
-                        if (users.contains("SPY") && !num.spy) {
-                            continue;
-                        }
-                        if (users.contains("GAPFUL") && !num.gapful) {
-                            continue;
-                        }
-                        if (users.contains("SUNNY") && !num.sunny) {
-                            continue;
-                        }
-                        if (users.contains("SQUARE") && !num.square) {
-                            continue;
-                        }
-                        if (users.contains("JUMPING") && !num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("HAPPY") && !num.happy) {
-                            continue;
-                        }
-                        if (users.contains("SAD") && !num.sad) {
-                            continue;
-                        }
-                        if (users.contains("-EVEN") && num.even) {
-                            continue;
-                        }
-                        if (users.contains("-ODD") && num.odd) {
-                            continue;
-                        }
-                        if (users.contains("-BUZZ") && num.buzz) {
-                            continue;
-                        }
-                        if (users.contains("-DUCK") && num.duck) {
-                            continue;
-                        }
-                        if (users.contains("-PALINDROMIC") && num.palindromic) {
-                            continue;
-                        }
-                        if (users.contains("-SPY") && num.spy) {
-                            continue;
-                        }
-                        if (users.contains("-GAPFUL") && num.gapful) {
-                            continue;
-                        }
-                        if (users.contains("-SUNNY") && num.sunny) {
-                            continue;
-                        }
-                        if (users.contains("-SQUARE") && num.square) {
-                            continue;
-                        }
-                        if (users.contains("-JUMPING") && num.jumping) {
-                            continue;
-                        }
-                        if (users.contains("-HAPPY") && num.happy) {
-                            continue;
-                        }
-                        if (users.contains("-SAD") && num.sad) {
-                            continue;
-                        }
-                        num.printShortInfo();
-                        count--;
-                    }
-                }
+        Set<String> allProperties =
```

Comment:
These 2 collection can be prepared during enum initialization, let's discuss.

Reply:
In this case we can create an isCorrect method in the enum Params.

Reply:
How to create one set AVAILABLE_PROPERTIES (both positive and negative)?
```
final static Set<String> POSITIVE_PROPERTIES = Stream.of(values()).map(Params::name).collect(Collectors.toSet());
final static Set<String> NEGATIVE_PROPERTIES =
                Stream.of(values()).map(Params::getNegative).collect(Collectors.toSet());
```

Reply:
flatMap

Reply:
Stream.of(values()).map(x -> Stream.of(x.name(), x.getNegative())).collect(Collectors.toSet())